modules/oracledb/oracledb.go

```diff
@@ -0,0 +1,657 @@
+package oracledb
+
+import (
+	"database/sql"
+	"strconv"
+	"strings"
+
+	"github.com/netdata/go.d.plugin/modules"
+
+	"gopkg.in/goracle.v2"
+)
+
+func init() {
+	creator := modules.Creator{
+		Create: func() modules.Module { return New("") },
+	}
+
+	modules.Register("oracledb", creator)
+}
+
+// OracleDB oracledb module.
+type OracleDB struct {
+	modules.Base
+	DSN string `yaml:"dsn,omitempty"`
+	db  *sql.DB
+}
+
+// New creates OracleDB mod.
+func New(dsn string) *OracleDB {
+	return &OracleDB{
+		DSN: dsn,
+	}
+}
+
+// Cleanup performs cleanup.
+func (m *OracleDB) Cleanup() {
+	err := m.db.Close()
+	if err != nil {
+		m.Errorf("cleanup: error on closing the oracle database [%s]: %v", m.DSN, err)
+	}
+}
+
+// Init makes initialization of the OracleDB mod.
+func (m *OracleDB) Init() bool {
+	if m.DSN == "" {
+		return false
+	}
+
+	// test the connectivity here.
+	if err := m.openConnection(); err != nil {
+		return false
+	}
+
+	// post Init debug info.
+	m.Debugf("using DSN [%s]", m.DSN)
+	return true
+}
+
+func (m *OracleDB) openConnection() error {
+	if m.db != nil {
+		if err := m.db.Ping(); err != nil {
+			m.db.Close()
+			m.db = nil
+
+			return m.openConnection()
```

Comment:
Just log the error and return nil. It will try to reconnect on every iteration, this is ok flow.

Reply:
and return an error ofc

Reply:
oh, i am wrong, your code is ok
sorry